pkg/loop/server.go

```diff
@@ -138,6 +174,12 @@ func (s *Server) Register(c services.HealthReporter) error { return s.checker.Re
 
 // Stop closes resources and flushes logs.
 func (s *Server) Stop() {
+	if s.dbStatsReporter != nil {
+		s.dbStatsReporter.Stop()
+	}
+	if s.db != nil {
+		s.Logger.ErrorIfFn(s.db.Close, "Failed to close database connection")
+	}
 	s.Logger.ErrorIfFn(s.checker.Close, "Failed to close health checker")
```

Comment:
sorry, is it clear why, for example, s.checker is not nil at this point, whereas s.db can still be nil and need to be checked?

Reply:
The checker is always present but the db is optional and so can be nil if not configured.

Reply:
- [ ] I'll make a note on the fields